GANDLF/utils/gandlf_logger.py

```diff
@@ -0,0 +1,44 @@
+import logging
+from logging import config
+import yaml
+from pathlib import Path
+
+
+def gandlf_logger_setup(
+    logger_name, config_path="logging_config.yaml"
+) -> logging.Logger:
+    """
+    It sets up the logger. Read from logging_config.
+
+    Args:
+        logger_name (str): logger name, the name should be the same in the logging_config
+        config_filename (str): file path for the configuration
+
+    Returns:
+        logging.Logger
+    """
+    with open(config_path, "r") as file:
+        config1 = yaml.safe_load(file)
+        logging.config.dictConfig(config1)
+
+    logging.captureWarnings(True)
+
+    return logging.getLogger(logger_name)
```

Comment:
I just noticed you read configuration file & re-setups logging config each time you need to get a new logger.
What do you think if we split this onto two calls?
- `gandlf_logger_setup` would return None, just explicitly sets up the configuration. We call it explicitly just once per every CLI entrypoint. IMO, the best places are:
  - https://github.com/mlcommons/GaNDLF/blob/new-apis_v0.1.0-dev/GANDLF/entrypoints/cli_tool.py#L27 for `gandlf` cli tool
  - https://github.com/mlcommons/GaNDLF/blob/new-apis_v0.1.0-dev/GANDLF/entrypoints/run.py#L186 `opd_way`  functions for every `gandlf_*` script
  
- The second part is just to use `logging.getLogger(???)` anywhere in the code to get a logger. No need to reconfigure anything. Also, thirdparty libraries would also be covered.

Reply:
@VukW I tried to implement this in a [new PR](https://github.com/mlcommons/GaNDLF/pulls#issue-2387298821)